src/collect/lift.jl

```diff
@@ -0,0 +1,88 @@
+# TODO: should we inline the following?
```

Comment:
I think we definitely should. Anyway the `return_type` will go away after inference in most cases, and we definitely don't want a function call just for the `if` branch.